osquery/killswitch/killswitch.cpp

```diff
@@ -0,0 +1,66 @@
+#include <osquery/flags.h>
+#include <osquery/killswitch.h>
+#include <osquery/registry_factory.h>
+#include <string>
+
+namespace osquery {
+
+FLAG(bool, enable_killswitch, true, "Enable killswitch plugin");
+FLAG(string,
+     killswitch_plugin,
+     "killswitch_filesystem",
+     "Killswitch plugin name.");
+
+Killswitch::Killswitch() {}
+Killswitch::~Killswitch() = default;
+
+Expected<bool, SwitchOnError> Killswitch::isSwitchOn(const std::string& key) {
+  PluginResponse response;
+  auto status = Registry::call("killswitch",
+                               {
+                                   {"action", "isEnabled"},
+                                   {"key", key},
+                               },
+                               response);
+  if (!status.ok()) {
+    return Error<SwitchOnError>(SwitchOnError::CallFailed, status.getMessage());
+  }
+
+  if (response.size() != 1) {
+    return Error<SwitchOnError>(
+        SwitchOnError::IncorrectResponseFormat,
+        "Response size should be 1 but is " + std::to_string(response.size()));
```

Comment:
You could use `<<`, if you rebase on current master. Your error could look like:
```c++
    return createError(
        SwitchOnError::IncorrectResponseFormat,
        "Response size should be 1 but is ") << response.size();
```
if you like it :)